### DISCUSSION TITLE
Ariadne Code Generator vs. sgqlc

### DISCUSSION THREAD

**Discussion by amacfie-tc** [mirumee/ariadne#1067] [Q&A] [answered]
The [sgqlc](https://github.com/profusion/sgqlc) project has a [code generator](https://github.com/profusion/sgqlc#code-generator) that appears to do something similar to [ariadne-codegen](https://github.com/mirumee/ariadne-codegen). What are the pros and cons of each tool?

**Comment by rafalp** [comment, marked as answer]
At quick glance they appear to treat GraphQL like database and implement and generate queries from custom query builder (like SQLAlchemy expressions).

Ariadne Codegen emulates WSDL approach of the yore giving you `GraphQLClient` class with methods corresponding to the GraphQL queries you've specified.
